### PR TITLE
Adding Figures to Introduction to C

### PR DESCRIPTION
**Pull request by bhuvy2**
## Description

Replacing a few Ascii Art images with some diagrams + SVG source.
Please review how the images look

## Related Issues

#12

## Checklist

Before you create this PR confirm that it meets all requirements listed below by checking the relevant checkboxes (`[x]`). This will ensure a smooth and quick review process.

- [x] I updated AUTHORS with my name and email.
- [x] I read the Contributor Guide and followed the process outlined there for submitting PRs.
- [x] My PR builds for *all* changed/updated/fixed behaviors (See [Contributor Guide]).
- [ ] I updated/added relevant documentation.
- [x] The title of the PR is descriptive and doesn't have [WIP]
- [ ] I updated CHANGELOG.md to add a description of the change.
- [x] I am willing to follow-up on review comments in a timely manner.

## Breaking Change

Does this cause any new material not previously covered in the course to be required?

- [ ] Yes, this is a breaking change (please indicate a breaking change in CHANGELOG.md).
- [x] No, this is *not* a breaking change.

### CODE REVIEW

**Comment by limjingrong** [comment]
I will uh eventually get to helping out with diagrams in the next like two weeks or so, sorry. Did agree to help with this last year.

File: introc/c_memory_model.tex [hunk 1]
```diff
@@ -56,37 +55,50 @@ \subsection{Structs}
...
-/*
-Currently, our memory looks like this with junk in those black spaces
-                ___ ___ ___ ___ ___ ___ ___ ___ ___ ___ ___
- bhuvan_name = |___|___|___|___|___|___|___|___|___|___|___|
+\end{lstlisting}
+
+Currently, our memory looks like the following image. There is nothing in those boxes
 
-*/
+\begin{figure}[H]
```
Comment (PradyumnaShome) [inline introc/c_memory_model.tex]: I don't know where we cover endianness, but we should mention that the figures are in big-endian and the addresses increase from left to right.

File: introc/c_memory_model.tex [hunk 2]
```diff
@@ -56,37 +55,50 @@ \subsection{Structs}
...
 strcmp(bhuvan_name->c_str, "bhuvan") == 0 //The strings are equal!
-\end{lstlisting}
+\end{minted}
 
-What that zero length array does is point to the \textbf{end of the struct} this means that the compiler will leave room for all of the elements calculated with respect to their size on the operating system (ints, chars, etc). The zero length array will take up no bytes of space.
+What that zero length array does is point to the \textbf{end of the struct} this means that the compiler will leave room for all of the elements calculated with respect to their size on the operating system (ints, chars, etc).
+The zero length array will take up no bytes of space.
 Since structs are just continuous pieces of memory, we can allocate \textbf{more} space than required and use the extra space as a place to store extra bytes.
 Although this seems like a parlor trick, it is an important optimization because to have a variable length string any other way, one would need to have two different memory allocation calls.
 This is highly inefficient for doing something as common in programming as is string manipulation.
```
Comment (PradyumnaShome) [inline introc/c_memory_model.tex]: Line 109: "when you have to an address from memory"  => "when you have to load an address from memory"

**Comment by bhuvy2** [comment]
What do you think of the pictures?

**Comment by PradyumnaShome** [comment]
I think they're helpful while trying to understand what's happening with memory when you declare structs. It's good to show them the example of length prefixed string as an alternative string implementation. We could certainly have more for the conceptual areas where code may not be most intuitive thing for a student to look at.

**Pull request merged**